cranelift/interpreter/src/step.rs

```diff
@@ -670,8 +670,24 @@ where
             ValueConversionKind::RoundNearestEven(ctrl_ty),
         )?),
         Opcode::Shuffle => unimplemented!("Shuffle"),
-        Opcode::Swizzle => unimplemented!("Swizzle"),
-        Opcode::Splat => unimplemented!("Splat"),
+        Opcode::Swizzle => {
+            let x = Value::into_array(&arg(0)?)?;
+            let s = Value::into_array(&arg(1)?)?;
+            let mut new = [0u8; 16];
+            for i in 0..new.len() {
+                if (s[i] as usize) < new.len() {
+                    new[i] = x[s[i] as usize];
```

Comment:
Could you return an error in case of going out-of-bounds?

Reply:
In the [docs](https://docs.rs/cranelift-codegen/0.76.0/cranelift_codegen/ir/trait.InstBuilder.html#method.swizzle), it's specified that the resulting lane should just be 0 for indices out of bounds - unless that's incorrect?

Reply:
Didn't know that. https://github.com/bytecodealliance/wasmtime/blob/16854e73c5bcd3323a51214b267f628ac41c9607/cranelift/codegen/src/isa/x64/lower.rs#L6339-L6343